### DISCUSSION TITLE
Change of Text by MQTT without HA?

### DISCUSSION THREAD

**Discussion by RaW79** [HASwitchPlate/openHASP#818] [Q&A] [answered]
Hi
I´m not using HomaAssitant so I´m reliant to send everything coming from outside by mqtt to the display which should be presented or updated.

From the example for[ Dash UI ](https://openhasp.haswitchplate.com/0.6/integrations/examples/example-dashui/) the soltution is given for an arc and a button.

For a button it´s:
```
hasp/plate35/command/p1b4.text %value%
```
But I do not realy like the button behaviour on a touch display for text. So I would like to use object type e.g. label for simple data presentation.

So e.g. for this Power Value:

```
{"id":15,"obj":"label","x":40,"y":370,"w":90,"h":90,"text":"1000W","text_font":32,"text_color":"#FFFFFF"}
```

May I kindly ask if this text "1000W" is changable by external mqtt publish? Can you inform how to publish to do?

Thank you very much!

**Comment by hb020** [comment]
Easy. Same as what you did with the button.

Python code here:

```python
    def sendProp(plate_name: str, el: str, prop: str, txt: str):
        """Send a property

        Args:
            plate_name (str): plate name
            el (str): element on screen (pXbY)
            prop (str): the property
            txt (str): text to send
        """
        if txt is None:
            txt = "[]"
        topic = f"hasp/{plate_name}/command/{el}.{prop}"
        mi = mqttc.publish(topic, txt)
        mi.wait_for_publish()  # this does not seem to block until all is gone!


    def sendTxt(plate_name: str, el: str, txt: str):
        """ send text to a label
        
        Args:
            plate_name (str): plate name
            el (str): element on screen (pXbY)
            txt (str): text to send
        """
        if txt is None:
            txt = "??"
        topic = f"hasp/{plate_name}/command/{el}.text"
        mi = mqttc.publish(topic, txt)
        mi.wait_for_publish()  # this does not seem to block until all is gone!
```

**Comment by fvanroie** [comment, marked as answer]
You can similarly use `hasp/plate35/command/p1b15.text` with payload `500W`

Or if you want to change multiple attributes at once: use `hasp/plate35/command/jsonl` with payload:
`{"page":1,"id":15,"text":"1125W",text_color":"#FF0000"}`

**Reply by RaW79** [reply]
Thanks, great also to have the option to change multiple attributes.
